lib/authorizer/digest.js

```diff
@@ -0,0 +1,94 @@
+var _extractNonce,
+    _extractRealm;
+
+/**
+ * Extracts Digest Auth nonce from a WWW-Authenticate header value.
+ * @private
+ */
+_extractNonce = function (string) {
+    // todo - make this more robust
```

Comment:
1 . Robust how?
2. Can be compacted as: `return string.match(/nonce.*"(.*)"/)[1]`.
3. This and the next function can be merged, with a second argument to indicate the control string, `nonce` or `realm`.